src/library/scala/collection/IndexedSeq.scala

```diff
@@ -47,15 +47,7 @@ trait IndexedSeqOps[+A, +CC[_], +C] extends Any with SeqOps[A, CC, C] { self =>
     s.asInstanceOf[S with EfficientSplit]
   }
 
-  override def reverseIterator: Iterator[A] = new AbstractIterator[A] {
-    private[this] var i = self.length
-    def hasNext: Boolean = 0 < i
-    def next(): A =
-      if (0 < i) {
-        i -= 1
-        self(i)
-      } else Iterator.empty.next()
-  }
+  override def reverseIterator: Iterator[A] = view.reverseIterator
```

Comment:
Maybe it's not used enough to matter, but are you sure this isn't going to result in a sizable performance hit?  The view reverse iterator does quite a bit more work, and may require double-bounds-checks for indexing.

Reply:
the view reverse iterator doesn't do that much more work. also, the regular iterator is already implemented by `view.iterator`, so I don't see it being any worse than that